src/Actions/main.vala

```diff
@@ -0,0 +1,44 @@
+#! /usr/bin/env -S vala workbench.vala --pkg libadwaita-1
+
+public void main() {
+    var demo = (Adw.StatusPage) workbench.builder.get_object("demo");
+
+    var demo_group = new SimpleActionGroup();
+    demo.insert_action_group("demo", demo_group);
+
+    // Action with no state or parameters
+    var simple_action = new SimpleAction("simple", null);
+    simple_action.activate.connect(() => {
+        message("%s action activated", simple_action.get_name());
+    });
+    demo_group.add_action(simple_action);
+
+    // Action with parameter
+    var bookmarks_action = new SimpleAction("open-bookmarks", new GLib.VariantType("s"));
+    bookmarks_action.activate.connect((action, parameter) => {
+        message("%s activated with %s", action.get_name(), parameter.get_string());
+    });
+    demo_group.add_action(bookmarks_action);
+
+    // Action with state
+    var toggle_action = new SimpleAction.stateful("toggle", null, new Variant.boolean(false));
+    toggle_action.notify.connect((action, state) => {
+        bool b = toggle_action.state.get_boolean();
+        message("%s action set to %s", toggle_action.name, b ? "true" : "false");
+    });
+
+    demo_group.add_action(toggle_action);
+
+    // Action with state and parameter
+    var scale_action = new SimpleAction.stateful("scale", new VariantType("s"), new Variant.string("100%"));
+
+    scale_action.notify.connect((action) => {
+        message("%s action set to %s", action.get_name(), scale_action.state.get_string());
```

Comment:
```suggestion
        message(@"$(action.name) action set to $(scale_action.state.get_string())");
```

Reply:
action and scale_action are here the same, aren't they? in that case I would only use one of them.

Reply:
Those are the same yes, but without using scale_action.state.get_string() I cannot access the string. Changing both to scale_action